mage_ai/data_cleaner/statistics/calculator.py

```diff
@@ -209,6 +213,40 @@ def __statistics_overview(self, series, col):
                     dates.sort_values().iloc[math.floor(len(dates) / 2)].isoformat()
                 )
                 data[f'{col}/min'] = dates.min().isoformat()
+            elif column_type == ColumnType.TEXT:
+                text_series = series_non_null
+                num_chars = text_series.apply(lambda string: len(string))
+                data[f'{col}/total_character_count'] = num_chars.sum()
+                data[f'{col}/avg_string_length'] = num_chars.mean()
+                text_series = text_series.str.replace(PUNCTUATION, ' ', regex=True)
+                text_series = text_series.str.lower()
+                text_series = text_series.str.split('\s+')
+                text_series = text_series.apply(
+                    lambda words: [word for word in words if word != '']
+                )
+                word_count = text_series.apply(lambda words: len(words))
+                data[f'{col}/total_word_count'] = word_count.sum()
+                data[f'{col}/avg_word_count'] = word_count.mean()
+                stop_words = text_series.apply(
+                    lambda words: reduce(
+                        lambda x, y: x + y,
+                        map(lambda word: 1 if word in STOP_WORD_LIST else 0, words),
+                    )
+                    if len(words)
+                    else 0
+                )
+                data[f'{col}/total_stop_word_count'] = stop_words.sum()
+                data[f'{col}/stop_word_rate'] = stop_words.sum() / data[f'{col}/total_word_count']
+                data[f'{col}/avg_stop_word_count'] = stop_words.mean()
+                data[f'{col}/word_distribution'] = (
+                    text_series.explode().value_counts().head(VALUE_COUNT_LIMIT).to_dict()
+                )
+            elif column_type == ColumnType.EMAIL:
+                valid_emails = series[~find_syntax_errors(series, ColumnType.EMAIL)]
```

Comment:
this method could be called twice. let's reduce duplicate calls

Reply:
Deduplicated call to `find_syntax_errors`